### PR TITLE
Netbox wait async

### PR DESCRIPTION
**Pull request by HustonMmmavr**

### CODE REVIEW

File: errors.lua [hunk 1]
```diff
@@ -382,6 +382,84 @@ local function netbox_call(conn, func_name, ...)
     )
 end
 
+
+local function _wrap_future_result(prefix, suffix, ...)
+    local args = {...}
+    local res, err = args[1], args[2]
+    if res ~= nil then
+        return res
+    end
```
Comment (olegrok) [inline errors.lua:391]: Seems you don't need varargs here. Just use `_wrap_future_result(prefix, suffix, res, err)`

Comment (HustonMmmavr) [reply errors.lua]: Thanks, fixed

File: errors.lua [hunk 2]
```diff
@@ -382,6 +382,85 @@ local function netbox_call(conn, func_name, ...)
...
+    elseif call_uri and type(call_uri) ~= 'string' then
+        error('Bad argument #3 to errors.netbox_wait_async_call' ..
+            ' (?string expected, got ' .. type(call_uri) .. ')', 2)
+    elseif call_func_name and type(call_func_name) ~= 'string' then
+        error('Bad argument #4 to errors.netbox_wait_async_call' ..
+        ' (?string expected, got ' .. type(call_func_name) .. ')', 2)
+    end
+
+    if future.method == 'eval' then
+        error("errors.netbox_wait_async_call doesn't support eval", 2)
```
Comment (rosik) [inline errors.lua:430]: Why? Are there any objective reasons?

Comment (HustonMmmavr) [reply errors.lua:430]: eval allows (without error):
```lua
conn:eval("return 1,2,3", nil)
```
and error handling will fail

**Pull request closed without merging**